Formula/ki.rb

```diff
@@ -0,0 +1,23 @@
+class Ki < Formula
+  desc "Kotlin Language Interactive Shell"
+  homepage "https://github.com/Kotlin/kotlin-interactive-shell"
+  url "https://github.com/Kotlin/kotlin-interactive-shell/archive/refs/tags/v0.3.3.tar.gz"
+  sha256 "46913b17c85711213251948342d0f4d0fec7dc98dd11c1f24eedb0409338e273"
+  license "Apache-2.0"
+  head "https://github.com/Kotlin/kotlin-interactive-shell.git", branch: "main"
+
+  depends_on "maven" => :build
+  depends_on "openjdk"
+
+  def install
+    system "mvn", "-DskipTests", "package"
+    libexec.install "lib/ki-shell.jar"
+    bin.write_jar_script libexec/"ki-shell.jar", "ki"
+  end
+
+  test do
+    output = pipe_output(bin/"ki")
```

Comment:
Shouldn't this be `shell_output`?

Reply:
I thought [you wanted me](https://github.com/Homebrew/homebrew-core/pull/74409#discussion_r606639048) to use `pipe_output`, but I'm happy to use either. This is a shell, so we'll need to close it, and it looks like `pipe_output` does it by default. `shell_output` requires sending the close command.

Reply:
I'm sorry for intervening, folks, but maybe it makes sense to perform test like this
```kotlin
    output = shell_output("#{bin}/ki.sh <<< '' 2>&1 | grep ki-shell")
    assert_match "ki-shell 0.3/1.4.32", output
```
This way it will not only check that shell works, but also will test a major version.

WDYT?

Reply:
I figured checking for `"ki-shell"` and `"Bye!"` would be sufficient because they won't (🤞) change between versions, so updating this Formula will be as easy as bumping the `url` and `sha256`. I'm open to whatever, but I think this current approach will require less maintenance in the long run.

Reply:
Oh, ok, makes sense!

Reply:
If you use `pipe_output` it should actually pipe something though, as I suggested earlier.

Reply:
If I use `shell_output`, the test just hangs.
```ruby
output = shell_output(bin/"ki")
```
I think it's because the `ki` shell is running and not terminating.

If I use `pipe_output`, it closes the `ki` shell, and I get the output.
```ruby
output = pipe_output(bin/"ki")
```
I can add arguments to `pipe_output`, but the `ki` shell doesn't use them.

I think `pipe_output` works because it [closes the `ki` shell](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula_assertions.rb#L79) while `shell_output` just [embeds the `ki` shell](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula_assertions.rb#L64) in a system command.

Which way should I write this line?

Reply:
in my example test does not hang, so you can reuse it and assert another things

Reply:
> in my example test does not hang, so you can reuse it and assert another things

My initial implementation was similar to your example, but [we decided `pipe_output` was a better direction](https://github.com/Homebrew/homebrew-core/pull/74409#discussion_r606191852).

Reply:
@SMillerDev, can I please get another look at this when you get a chance?

Reply:
While we do have a few tests where `pipe_output` is used without really piping anything, I think it's preferable to just do this:
```ruby
test do
  output = pipe_output(bin/"ki", ":q")
  assert_match "ki-shell", output
  assert_match "Bye!", output
end
```

Reply:
Sure 👍